iceoryx_utils/include/iceoryx_utils/internal/cxx/string_internal.hpp

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef IOX_UTILS_CXX_STRING_INTERNAL_HPP
+#define IOX_UTILS_CXX_STRING_INTERNAL_HPP
+
+#include <cstdint>
+#include <cstring>
+#include <string>
+
+namespace iox
+{
+namespace cxx
+{
+template <uint64_t>
+class string;
+
+namespace internal
+{
+template <uint64_t N>
+using charArray = char[N];
+
+/// @brief struct to get capacity of fixed string/string literal
+/// @note capa is a dummy value for any type other than cxx::string and char
+template <typename T>
+struct GetCapa
+{
+    static constexpr uint64_t capa = 0U;
+};
+
+template <uint64_t N>
+struct GetCapa<string<N>>
+{
+    static constexpr uint64_t capa = N;
+};
+
+template <uint64_t N>
+struct GetCapa<char[N]>
+{
+    static constexpr uint64_t capa = N - 1U;
+};
+
+/// @brief struct to get size of fixed string/string literal/std::string
+template <typename T>
+struct GetSize;
+
+template <uint64_t N>
+struct GetSize<string<N>>
+{
+    static uint64_t call(const string<N>& data)
```

Comment:
please add noexept to all call() functions

Reply:
done